### PR TITLE
Review: Simplify layout handling

### PR DESCRIPTION
**Pull request by ghengeveld**
## What I did

Review mode no longer collapses or restores sidebar/panel visibility when entering or leaving a review; it still snapshots and restores sidebar filters only.

On the review summary page and on story routes with a `collection` query param, the sidebar is not rendered (including the mobile menu drawer), sidebar keyboard shortcuts (`toggleNav`, `focusNav`, `search`) are ignored, and the toolbar “Show sidebar” control is hidden. The review toolbar header on story canvases only renders when `collection` is present.

The fixed-position review summary portal no longer reserves left nav width on review routes, so the summary fills the full width when the sidebar is suppressed.

#### The changes in this PR are covered in the following automated tests:

- [x] stories
- [x] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

> [!CAUTION]
> This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!

1. Run the internal Storybook UI (`cd code && yarn storybook:ui`).
2. Trigger or open a review summary (`?path=/review/`).
3. Confirm the summary is full width with no empty gutter where the sidebar would be, and that the sidebar (desktop) / navigation menu (mobile) is not shown.
4. Open a reviewed story (`collection` query param present): confirm no sidebar, no review toolbar header on normal stories without `collection`, review header visible with `collection`, and no “Show sidebar” toolbar button.
5. Enter and exit review mode and confirm sidebar/panel visibility matches what you had before entering; only sidebar filters should change during review.

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli-storybook/src/sandbox-templates.ts`
- [ ] Declare whether manual QA will be needed for this PR during the next release, through `qa:needed` or `qa:skip`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

  - `bug`: Internal changes that fixes incorrect behavior.
  - `maintenance`: User-facing maintenance tasks.
  - `dependencies`: Upgrading (sometimes downgrading) dependencies.
  - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
  - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
  - `documentation`: Documentation **only** changes. Will not show up in release changelog.
  - `feature request`: Introducing a new feature.
  - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
  - `other`: Changes that don't fit in the above categories.

   </details>

### 🦋 Canary release

This pull request has been released as version `0.0.0-pr-35329-sha-ec096ee6`. Try it out in a new sandbox by running `npx storybook@0.0.0-pr-35329-sha-ec096ee6 sandbox` or in an existing project with `npx storybook@0.0.0-pr-35329-sha-ec096ee6 upgrade`.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-35329-sha-ec096ee6`](https://npmjs.com/package/storybook/v/0.0.0-pr-35329-sha-ec096ee6) |
| **Triggered by** | @ghengeveld |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`review-sidebar-layout`](https://github.com/storybookjs/storybook/tree/review-sidebar-layout) |
| **Commit** | [`ec096ee6`](https://github.com/storybookjs/storybook/commit/ec096ee67c98a5e6a9a335c7f2b4b50eac5db5b1) |
| **Datetime** | Tue Jun 30 11:26:06 UTC 2026 (`1782818766`) |
| **Workflow run** | [28440764779](https://github.com/storybookjs/storybook/actions/runs/28440764779) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/publish.yml) or locally with `gh workflow run --repo storybookjs/storybook publish.yml --field pr=35329`_
</details>

## Summary by CodeRabbit

* **New Features**
  * Review-collection routes (story + review query param) are now recognized to tailor toolbar/menu visibility and navigation behavior.
* **Bug Fixes**
  * Sidebar and navigation shortcuts are blocked on review-manager views to prevent unintended focus/search/toggles.
  * Sidebar/chrome insets, fullscreen toolbar visibility, and mobile navigation menu rendering now adapt to review-manager layout; mobile menu can be programmatically hidden.
  * Review mode and leaving-review flows are filter-focused, with refined auto-accept/notification behavior, improved route-aware notification handling, and a navigation-in-flight guard to prevent premature re-entry.
* **Tests**
  * Updated/added stories and test coverage for route detection, filter restoration, notification dismissal, and review navigation edge cases.

### CODE REVIEW

File: code/core/src/manager/components/review/review-mode.ts [hunk 1]
```diff
@@ -51,37 +61,67 @@ const readJson = <T>(key: string): T | null => {
...
+/** Snapshot sidebar filters once per browser session (no narrowing). */
+export const initializeSessionFiltersIfNeeded = (
+  api: ReviewModeApi,
+  filters: ReviewModeFilters
+): void => {
+  if (sessionStore.read(FILTERS_SNAPSHOT_SESSION_KEY) !== null) {
+    return;
+  }
+
+  sessionStore.write(FILTERS_SNAPSHOT_SESSION_KEY, JSON.stringify(filters));
```
[the remark this answers is not in the document]
Comment (ghengeveld) [reply code/core/src/manager/components/review/review-mode.ts]: The chrome and filter snapshots are intentionally scoped to the browser session, not a single review cycle. `CHROME_INITIALIZED_SESSION_KEY` ensures collapse only runs on the first review-page visit per session, and the stored snapshot is the pre-review state we restore on exit. Clearing the snapshot keys on exit would break that session-scoped contract and the existing tests in `review-mode.test.ts`.

**Comment by ghengeveld** [comment]
Babysit update while AFK:

- **Fixed CI failure** (`typescript-validation`): `deriveViewMode` in `ReviewWidget.stories.tsx` now handles `parsePath().viewMode` being `string | undefined` (`95a143d`).
- **Earlier fix** (`2891e6b`): clear chrome/filter session snapshots on `exitReviewMode` so later exits cannot restore stale state.

All CodeRabbit threads are resolved. Local validation: `core:check`, review unit tests, and ReviewWidget/ReviewPage story tests pass.

Awaiting CI re-run and @yannbf review.

**Review by yannbf — APPROVED** [review]
LGTM as long as the flicker is resolved. The story has to be unmounted so there's no flicker in the navigation
